selfdrive/camerad/cameras/camera_qcom2.cc

```diff
@@ -76,12 +76,36 @@ int cam_control(int fd, int op_code, void *handle, int size) {
   return ret;
 }
 
+std::optional<int32_t> device_acquire(int fd, int32_t session_handle, void *data) {
+  struct cam_acquire_dev_cmd cmd = {
+      .session_handle = session_handle,
+      .handle_type = CAM_HANDLE_USER_POINTER,
+      .num_resources = (uint32_t)(data ? 1 : 0),
+      .resource_hdl = (uint64_t)data,
+  };
+  int err = cam_control(fd, CAM_ACQUIRE_DEV, &cmd, sizeof(cmd));
+  return err == 0 ? std::make_optional(cmd.dev_handle) : std::nullopt;
+};
+
+int device_config(int fd, int32_t session_handle, int32_t dev_handle, uint64_t packet_handle) {
+  struct cam_config_dev_cmd cmd = {
+      .session_handle = session_handle,
+      .dev_handle = dev_handle,
+      .packet_handle = packet_handle,
+  };
+  return cam_control(fd, CAM_CONFIG_DEV, &cmd, sizeof(cmd));
+}
+
 int device_control(int fd, int op_code, int session_handle, int dev_handle) {
-  // start stop and release are all the same
-  static struct cam_release_dev_cmd release_dev_cmd;
-  release_dev_cmd.session_handle = session_handle;
-  release_dev_cmd.dev_handle = dev_handle;
-  return cam_control(fd, op_code, &release_dev_cmd, sizeof(release_dev_cmd));
+  if (op_code == CAM_START_DEV || op_code == CAM_STOP_DEV) {
```

Comment:
Think the old code was fine since all the structs are the same, I was just wondering why it was changed with respect to the original code.

Reply:
so how about using cam_start_stop_dev_cmd ? release is a bit confusing.

Reply:
ah, I guess. If you think that's clearer it's fine